src/Contract/Crypto/ECDSA.purs

```diff
@@ -0,0 +1,78 @@
+-- | A module that implements crypto primitives that match CIP-49 SECP256k1
+-- | ECDSA spec.
+module Contract.Crypto.ECDSA
+  ( verifyEcdsaSecp256k1Signature
+  , signEcdsaSecp256k1
+  , deriveEcdsaSecp256k1PublicKey
+  , module X
+  , mkECDSAPublicKey
+  , unECDSAPublicKey
+  , mkMessageHash
+  , unMessageHash
+  ) where
+
+import Prelude
+
+import Ctl.Internal.Types.ByteArray (ByteArray)
+import Data.Maybe (Maybe)
+import Data.Newtype (unwrap, wrap)
+import Effect.Aff (Aff)
+import Noble.Secp256k1.ECDSA
+  ( ECDSAPublicKey
+  , ECDSASignature
+  , MessageHash
+  , PrivateKey
+  ) as X
+import Noble.Secp256k1.ECDSA
+  ( ECDSAPublicKey
+  , ECDSASignature
+  , MessageHash
+  , PrivateKey
+  , signECDSA
+  , verifyECDSA
+  )
+import Noble.Secp256k1.ECDSA
+  ( getECDSAPublicKey
+  , mkECDSAPublicKey
+  , mkMessageHash
+  , unECDSAPublicKey
+  , unMessageHash
+  ) as ECDSA
+
+-- | Verify ECDSA signature for a message hash.
+-- |
+-- | This function matches CIP-49 definition:
+-- |
+-- | https://github.com/cardano-foundation/CIPs/blob/master/CIP-0049/README.md
+verifyEcdsaSecp256k1Signature
+  :: ECDSAPublicKey -> MessageHash -> ECDSASignature -> Boolean
+verifyEcdsaSecp256k1Signature publicKey messageHash signature =
+  verifyECDSA signature messageHash publicKey
+
+-- | Sign a message hash with a private key, producing a signature compatible
+-- | with `verifyEcdsaSecp256k1Signature`.
+signEcdsaSecp256k1 :: PrivateKey -> MessageHash -> Aff ECDSASignature
```

Comment:
If there is both sign and verify functions I think it would be good if there was joint test, just to check that they work.